### PR TITLE
ARGO-321 Implement Topics Resource 

### PR DESCRIPTION
**Pull request by kaggis**
- [X] Topic Resource Model Implementation
- [X] Handlers for topic related api calls

### CODE REVIEW

**Comment by kaggis** [comment]
please review 
/cc @pkoro @dpavlos

File: argo-messaging.spec [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#debuginfo not supported with Go
+%global debug_package %{nil}
+
+Name: argo-messaging
+Summary: ARGO Messaging API for broker network
+Version: 0.9.0
```
Comment (pkoro) [inline argo-messaging.spec]: Change version to `0.9.2` to match with the changelog.

File: argo-messaging.spec [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Release: 1%{?dist}
+License: ASL 2.0
+Buildroot: %{_tmppath}/%{name}-buildroot
+Group:     EGI/SA4
```
Comment (pkoro) [inline argo-messaging.spec]: I think this is a remnant we have in the argo-web-api as well. I would suggest better to use

```
Group: Applications/Publishing
```

What do others think?

File: argo-messaging.spec [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Source0: %{name}-%{version}.tar.gz
+BuildRequires: golang
+BuildRequires: bzr
```
Comment (pkoro) [inline argo-messaging.spec]: I am not sure if we need bzr for this package. It seems we need `godep` or the package that provides `godep` if it's not part of `golang`

File: argo-messaging.spec [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+BuildRequires: git
+ExcludeArch: i386
+Obsoletes: argo-messaging
```
Comment (pkoro) [inline argo-messaging.spec]: Remove this line.

File: argo-messaging.spec [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+export GOPATH=$PWD
+cd src/github.com/ARGOeu/argo-messaging/
+go get github.com/tools/godep
+godep restore
+godep update ...
+go install
+
+%install
+%{__rm} -rf %{buildroot}
+install --directory %{buildroot}/var/www/argo-web-api
```
Comment (pkoro) [inline argo-messaging.spec]: Change `argo-web-api` to `argo-messaging`.

File: argo-messaging.spec [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+install --mode 755 bin/argo-messaging %{buildroot}/var/www/argo-messaging/argo-messaging
+
+install --directory %{buildroot}/etc
+install --mode 644 src/github.com/ARGOeu/argo-messaging/config.json %{buildroot}/etc/argo-messaging/config.json
+
+install --directory %{buildroot}/etc/init
+install --mode 644 src/github.com/ARGOeu/argo-messaging/argo-messaging.conf %{buildroot}/etc/init/
```
Comment (pkoro) [inline argo-messaging.spec]: This file (`argo-messaging.conf`) is not part of this PR. Better add an init script configuration file as part of this PR.

File: argo-messaging.spec [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+%attr(0755,root,root) /var/www/argo-messaging/argo-messaging
+%attr(0644,root,root) /etc/argo-messaging/config.json
+%attr(0644,root,root) /etc/init/argo-messaging.conf
+
+%changelog
+* Thu Jan 21 2015 Konstantinos Kagkelidis <kaggis@gmail.com> - 0.9.2-1%{?dist}
+- ARGO-320 Implement Message Resource Model Add Unittests
+* Wed Jan 20 2015 Konstantinos Kagkelidis <kaggis@gmail.com> - 0.9.1-1%{?dist}
+- ARGO-319 Add initial api frontend Add brokers package with kafka connector implementation
+* Wed Jan 20 2015 Christos Kanellopoulos <skanct@gmail.com> - 0.9.0-1%{?dist}
```
Comment (pkoro) [inline argo-messaging.spec]: Change 2015 to 2016 here and above in the changelog section.

File: doc/swagger/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
+# ARGO Messaging API expressed in swagger
+swagger: '2.0'
+info:
+  title: argo-messaging-api
+  description: Argo api for publishing / consuming messages from broker network
+  version: "0.9.0"
```
Comment (pkoro) [inline doc/swagger/swagger.yaml]: Change version number here also.

File: main.go [hunk 9]
```diff
@@ -8,18 +8,19 @@ import (
...
 
-	"github.com/ARGOeu/argo-messaging/config"
-	"github.com/gorilla/mux"
+	"github.com/ARGOeu/argo-messaging/Godeps/_workspace/src/github.com/gorilla/mux"
 )
 
 func main() {
 	defer broker.CloseConnections()
-	broker.Initialize(config.Kafka.Server)
+	//broker.Initialize(kafkaCfg.Server)
```
Comment (pkoro) [inline main.go]: Not sure whether this should be commented out here.

**Comment by pkoro** [comment]
I think we also need a `Makefile` for this PR on the top level directory (only for rpm building).

File: argo-messaging.spec (hunk 3 above)
Comment (kaggis) [inline argo-messaging.spec]: Aggreed the bzr requirement was needed for labix/mgo driver but i've checked and now it's available as a github repo

File: argo-messaging.spec (hunk 6 above)
Comment (kaggis) [inline argo-messaging.spec]: File has been placed

**Comment by kaggis** [comment]
- Fixes on the issues commented have been made (spec file, main.go)
- Makefile has been Added

please review
/cc @pkoro

File: README.md [hunk 10]
```diff
@@ -1,5 +1,46 @@
...
+
+4. Get dependencies:
+
+   Argo-messaging uses godep tool for dependency handling.
+   To install godep tool issue:
+
+      `go get github.com/tools/godep`
+      `godep update ...`
+      `godep restore`
+
```
Comment (pkoro) [inline README.md]: I tried to follow this step by step instructions. I think that steps 3 and 4 should be changed in sequence. 

Also I see the update & restore commands are the other way around in the travis and spec files. Should this be the case also here?

File: argo-messaging.spec [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+Version: 0.9.1
+Release: 1%{?dist}
+License: ASL 2.0
+Buildroot: %{_tmppath}/%{name}-buildroot
+Group: Unspecified
+Source0: %{name}-%{version}.tar.gz
+BuildRequires: golang
+BuildRequires: git
+ExcludeArch: i386
+Obsoletes: argo-messaging
```
Comment (pkoro) [inline argo-messaging.spec]: Remove the `Obsoletes` line.

File: doc/swagger/swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,67 @@
+# ARGO Messaging API expressed in swagger
+swagger: '2.0'
+info:
+  title: argo-messaging-api
+  description:  ARGO Messaging API for broker network
+  version: "0.9.0-1"
```
Comment (pkoro) [inline doc/swagger/swagger.yaml]: Version mismatch. It should be `0.9.1`

File: argo-messaging.spec [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+Installs the ARGO Messaging API
+
+%prep
+%setup
+
+%build
+export GOPATH=$PWD
+cd src/github.com/ARGOeu/argo-messaging/
+go get github.com/tools/godep
+godep restore
```
Comment (pkoro) [inline argo-messaging.spec]: You need to add the following line right after installing godep and before calling it in subsequent lines:

```
export PATH=$PATH:$GOPATH/bin
godep restore
...
```

**Comment by pkoro** [comment]
+1
